drivers/flash/flash_stm32l5_u5.c

```diff
@@ -387,7 +387,7 @@ void flash_stm32_page_layout(const struct device *dev,
 		if (stm32_flash_layout[0].pages_count == 0) {
 			if ((regs->OPTR & FLASH_STM32_DBANK) == FLASH_STM32_DBANK) {
 				/* flash with dualbank has 2k pages */
-				stm32_flash_layout[0].pages_count = FLASH_PAGE_NB;
+				stm32_flash_layout[0].pages_count = FLASH_SIZE / FLASH_PAGE_SIZE;
```

Comment:
comment to update : stm32U5 page size is _not_ 2KB

Reply:
This should be resolved the second commit.
``` C
if ((regs->OPTR & FLASH_STM32_DBANK) == FLASH_STM32_DBANK) {
	/* L5 flash with dualbank has 2k pages */
	/* U5 flash pages are always 8 kB in size */
	stm32_flash_layout[0].pages_count = FLASH_SIZE / FLASH_PAGE_SIZE;
	stm32_flash_layout[0].pages_size = FLASH_PAGE_SIZE;
```